Assets/Scripts/Spawn System/PlayerRespawn.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+
+//attach to player
+public class PlayerRespawn : MonoBehaviour
+{
+    GameObject player;
+    SpawnPoint spawn;
```

Comment:
can change this to a serializefield containing a checkpoint (see recommended changes in SpawnPoint)